toolchain/check/context.h

```diff
@@ -45,6 +46,25 @@ struct LookupResult {
   SemIR::InstId inst_id;
 };
 
+// Information about an access.
+struct AccessInfo {
+  // The instruction being accessed.
+  SemIR::InstId base_id;
+
+  // The highest allowed access for a lookup.
+  //
+  // - If this is set to `AccessKind::Private` then this access will be
+  // allowed for `AccessKind::Private`, `AccessKind::Protected` and
+  // `AccessKind::Public`.
+  //
+  // - If this is set to `AccessKind::Protected` then this will be
+  // allowed for both `AccessKind::Protected` and `AccessKind::Public`.
+  //
+  // - If this is set to `AccessKind::Public` then this will only be
+  // allowed for `AccessKind::Public`.
```

Comment:
```suggestion
  // The highest allowed access for a lookup. For example, `Protected` allows
  // access to `Public` and `Protected` names, but not `Private`.
```

The current comment seems a little long, does this shorter example work for you?

Reply:
Sounds good.